src/components/NcAppSidebar/NcAppSidebar.vue

```diff
@@ -451,177 +448,265 @@ export default {
+	components: {
+		Cog,
+	},
+	data() {
+		return {
+			showSidebar: true,
+		}
+	},
+}
+</script>
+<style scoped>
+/* This styles just mock NcContent and NcAppContent */
+.content-styleguidist {
+	position: relative !important;
+	/* Just to prevent jumping when the sidebar is hidden */
+	min-height: 360px;
+}
+
+.main-content {
+	position: absolute;
+	height: 100%;
+	width: 100%;
+}
+
+/* Fix styles on this style guide page */
+@media only screen and (max-width: 512px) {
+	:deep(aside) {
+		width: calc(100vw - 64px) !important;
+	}
+}
+
+.top-bar {
+	display: flex;
+	justify-content: flex-end;
+	/* preserve space for toggle button */
+	padding-inline-end: var(--app-sidebar-offset);
+	/* same as on toggle button, but doesn't have to be the same */
+	margin: var(--app-sidebar-padding);
+}
+</style>
+```
+</docs>
 
-							<!-- name -->
-							<div class="app-sidebar-header__name-container">
-								<div class="app-sidebar-header__mainname-container">
-									<!-- main name -->
-									<h2 v-show="!nameEditable"
-										:id="`app-sidebar-vue-${uid}__header`"
-										ref="header"
-										v-linkify="{text: name, linkify: linkifyName}"
-										:aria-label="title"
-										:title="title"
-										class="app-sidebar-header__mainname"
-										:tabindex="nameEditable ? 0 : -1"
-										@click.self="editName">
-										{{ name }}
-									</h2>
-									<template v-if="nameEditable">
-										<form v-click-outside="() => onSubmitName()"
-											class="app-sidebar-header__mainname-form"
-											@submit.prevent="onSubmitName">
-											<input ref="nameInput"
-												v-focus
-												class="app-sidebar-header__mainname-input"
-												type="text"
-												:placeholder="namePlaceholder"
-												:value="name"
-												@keydown.esc.stop="onDismissEditing"
-												@input="onNameInput">
-											<NcButton type="tertiary-no-background"
-												:aria-label="changeNameTranslated"
-												native-type="submit">
-												<template #icon>
-													<ArrowRight :size="20" />
-												</template>
-											</NcButton>
-										</form>
+<template>
+	<transition appear
+		name="slide-right"
+		@before-enter="onBeforeEnter"
+		@after-enter="onAfterEnter"
+		@before-leave="onBeforeLeave"
+		@after-leave="onAfterLeave">
+		<aside v-show="open"
+			id="app-sidebar-vue"
+			ref="sidebar"
+			class="app-sidebar"
+			:aria-labelledby="`app-sidebar-vue-${uid}__header`"
+			@keydown.esc="onKeydownEsc">
+			<!--
+				We cannot render toggle button inside sidebar (aside#app-sidebar-vue), because it is hidden then the toggle is needed.
+				But we also need transition with the sidebar to be the root of this component to use it as a single UI element, allowing to use `v-show`.
+				So we cannot render the toggle button directly in this component.
+				As a simple solution - render it in the content to keep correct position.
+			-->
+			<Teleport selector="#content-vue">
```

Comment:
We should add `no-toggle` prop to allow hiding the toggle button.

Reply:
but unrelated, just to note.

Reply:
I can restore a PR with a feature of placing `NcAppSidebarToggle` button anywhere and `no-toggle` prop.

Reply:
Only `no-toggle`: https://github.com/nextcloud-libraries/nextcloud-vue/pull/5702